src/renderer/store/lobby.store.ts

```diff
@@ -0,0 +1,288 @@
+    try {
+        battleActions.resetToDefaultBattle(undefined, undefined, undefined, true);
+        battleStore.isOnline = true;
+        lobbyStore.error = undefined;
+        const response = await window.tachyon.request("lobby/create", data);
+        console.log("Tachyon: lobby/create:", response.status, response.data);
+        lobbyStore.activeLobby = parseLobbyResponseData(response.data); //Set the active lobby data first...
+        battleStore.isLobbyOpened = true;
+        //Note, we don't do any 'user/subscribeUpdates' here because when we create it there is only ourselves on initial join.
+        //If that ever changes, if a whole party joins instantly for example, then we need to revisit this.
+    } catch (error) {
+        console.error("Error with request lobby/create", error);
+        lobbyStore.error = "Error with request lobby/create";
+    }
+}
+
+async function joinLobby(id: LobbyJoinRequestData) {
+    try {
+        battleActions.resetToDefaultBattle(undefined, undefined, undefined, true);
+        battleStore.isOnline = true;
+        lobbyStore.error = undefined;
+        const response = await window.tachyon.request("lobby/join", id);
+        console.log("Tachyon: lobby/join:", response.status, response.data);
+        lobbyStore.activeLobby = parseLobbyResponseData(response.data);
+        battleStore.isLobbyOpened = true;
+        //Subscribe to obtain user data
+        const userSubList: UserId[] = [];
+        for (const memberKey in lobbyStore.activeLobby.members) {
+            if (lobbyStore.activeLobby.members[memberKey]) {
+                userSubList.push(lobbyStore.activeLobby.members[memberKey].id);
+            }
+        }
+        window.tachyon.request("user/subscribeUpdates", { userIds: userSubList });
+    } catch (error) {
+        console.error("Error with request lobby/join", error);
+        lobbyStore.error = "Error with request lobby/join";
+    }
+}
+
+// We use this function to normalize both LobbyCreateRequestData and LobbyJoinRequestData into the Lobby type for use in the renderer
+function parseLobbyResponseData(data: LobbyCreateOkResponseData | LobbyJoinOkResponseData) {
+    //Do some cleanup in case there's old data in the stores.
+    battleStore.battleOptions.mapOptions.customStartBoxes = [];
+
+    //Set up a basic object to hold the data
+    const lobbyObject: Lobby = {
+        id: data.id,
+        name: data.name,
+        playerCount: 0, //Placeholder value; We can count this from the entries below in the team config
+        maxPlayerCount: 0, //Placeholder value; Ditto for calculating max players
+        mapName: data.mapName,
+        engineVersion: data.engineVersion,
+        gameVersion: data.gameVersion,
+        allyTeams: {},
+        members: {},
+    };
+    battleStore.battleOptions.engineVersion = data.engineVersion;
+    battleStore.battleOptions.gameVersion = data.gameVersion;
+    db.maps.get(data.mapName).then((map) => {
```

Comment:
Assert for when it's not found?

Overall, the more I stare at the map store the more I think we need to attempt to rework it again to have better separation from live maps list and not have all tied to it... But that's not for this PR...

Reply:
We could do something here, yes. If we somehow create or join a lobby and get the success response with an unidentified map name, then the only reasonable response is to bail on the lobby altogether? It should be pretty rare, I hope.
Do we have a way to essentially say “I think my maps list may be inaccurate or out of date, let’s refresh it”?

Reply:
This is resolved. If the `db.maps.get()`` returns ``undefined``, a console error and notification will be generated. The user will still join the lobby, however. Since an ``undefined`` mapname cannot be downloaded, instead the user will see a disabled "Start the game" button and a blank map preview.
To resolve, they'll have to leave the lobby. Once all users leave, the lobby, along with the undefined mapname, will be disposed of. Ideally, proper lobby creation will prevent this situation from happening, but at least it will be handled gracefully now.